mvrx/src/main/kotlin/com/airbnb/mvrx/RealMvRxStateStore.kt

```diff
@@ -120,10 +187,13 @@ class RealMvRxStateStore<S : Any>(initialState: S) : MvRxStateStore<S> {
     }
 
     private fun flushSetStateQueue() {
```

Comment:
@hellohuanlin isn't this called `flushQueues()` which does drain the setState queue. Not sure what is changing here. Could you clarify?

Reply:
let's say we have `[G1, G2, G3]` in `getStateQueue` and `[S1, S2, S3]` in `setStateQueue`. we want to purge all `[S1, S2, S3]` before taking any task in `getStateQueue`. With the original implementation, the execution order will be `S3 -> G3 -> S2 -> G2 -> S1 -> G1`, which has the race condition. With this change, the execution order will be `S3 -> S2 -> S1 -> G3 -> G2 -> G1`

Reply:
Oh, actually if you follow the execution path, the original implementation actually may have missing reducers. 

Let's say we have `[G1]` and `[S1, S2, S3]`, in the original implementation, execution order will be `S1 -> G1 -> S2` (`S3` is missing)

So this PR not just fix the race condition, but also fix the missing reducer.

Reply:
I will update the PR description